public/scripts/tagsList.js

```diff
@@ -1,21 +1,148 @@
-function getTagObject(content){
+let labelArray = null
+let currentFocus = -1
+
+function findLabelSuggestion(input) {
+    if (input === '')
+        return []
+    let suggestion = []
+
+    suggestion = labelArray.filter(element => {
+        if (input.length > element.length)
+            return false
+        return input.toUpperCase() === element.substring(0, input.length).toUpperCase()
+    })
+
+    return suggestion
+}
+
+function createTagObject(content) {
     let template = document.createElement('template')
-    template.innerHTML = 
-    `<div class="tag-element">
+    template.innerHTML =
+        `<div class="tag-element">
         <span class="tag-element-text" data-value=${content}>${content}</span> 
-        <button class="delete-tag-button"></button> 
+        <button class="delete-tag-button" type="button">x</button> 
     </div>`
+    template.content.querySelector('button').onclick = (event) => {
+        event.target.parentNode.remove()
+    }
     return template.content  // this is so much simple :O
+}
+
+function closeRecommendationList() {
+    let lists = document.querySelectorAll('.autocomplete-list')
+    if (lists) {
```

Comment:
querySelectorAll _[always returns a truthy value](https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelectorAll)_: a NodeList (an collection of sorts)

Try `document.querySelectorAll('bunica')` in the console right now.

What you want to be checking is the _length_ of the returned array. That's `0` when nothing is found.

So use `if (lists.length)`.

Reply:
I will remove it altogether since It's redundant